src/content/blog/astro-480.mdx

```diff
@@ -0,0 +1,177 @@
+
+## Experimental: Request Rewriting
+
+Astro 4.8 introduces experimental support for Request rewriting. Rewriting is a routing feature that allows you to show the content of a different page from the current URL. This can be useful for showing different content based on the user's location, device, or other conditions without needing multiple different URLs. This feature is sometimes also called "rerouting" in other frameworks.
+
+To enable this feature, configure the `experimental.rewriting` option in your Astro config:
+
+```js
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  experimental: {
+    rewriting: true,
+  },
+});
+```
+
+From your pages and endpoints, you can then access a `rewrite()` method on the Astro global or context object:
+
+```astro
+---
+if (!Astro.props.allowed) {
+	return Astro.rewrite("/")
+}
+---
+```
+
+```ts
+export function GET(ctx) {
+  if (!ctx.locals.allowed) {
+    return ctx.rewrite("/");
+  }
+}
+```
+
+This feature can also be used inside middlewares, by passing a parameter to the `next()` method with the same type as the `rewrite` method.
+
+```ts
+export function onRequest(ctx, next) {
+  if (!ctx.cookies.get("allowed")) {
+    return next("/");
+  }
+  return next();
+}
+```
+
+For more information on rewriting, visit [the experimental rewriting docs](https://docs.astro.build/en/reference/configuration-reference/#experimentalrewriting).
+
+## Performance improvements
+
+Thanks to the recent hard work from [@bluwy](https://github.com/bluwy) both across the JavaScript ecosystem and in Astro itself, Astro is faster than ever! This release includes a number of performance improvements, cutting build and rendering times across the board.
+
+In our own benchmarks, we've observed build times improve by up to 20% in specific cases. No changes are needed to take advantage of these improvements, just upgrade to Astro 4.8 and enjoy the faster builds!
+
+## Ability to define multiple routes with the same entrypoint
+
+Astro integration authors discovered that trying to inject multiple routes with the same entrypoint using the [`injectRoute`](https://docs.astro.build/en/reference/integrations-reference/#injectroute-option) function would result in an unintended behavior at build time where the last injected route would override the previous ones.
+
+This issue has existed since the introduction of the `injectRoute` function, almost two years ago (in Astro 1.0), but thanks to a contribution by [@goulvenclech](https://github.com/goulvenclech), it is now finally fixed in Astro 4.8!
+
+With our thriving developer community and so many new integrations being built all the time, it's important to us that Astro is a great platform for building integrations and we are pleased that this paves the way for even more integrations in the future!
```

Comment:
```suggestion
With our thriving developer community and so many new integrations being built all the time, it's important to us that Astro is a great platform to build for and we are pleased that this paves the way for even more integrations in the future!
```
Sounds great! To avoid repeating integrations so many times, *could* replace with:

"Astro is a great platform to build for"
"Astro is a great platform to build on top of"
"Astro is a great platform for builders"
etc. 
Whatever sounds right to you!